source/traction_boundary_conditions/lithostatic_pressure.cc

```diff
@@ -0,0 +1,469 @@
+          spherical_representative_point[1] *= degrees_to_radians;
+          spherical_representative_point[2] *= degrees_to_radians;
+        }
+      else if (const GeometryModel::Box<dim> *gm = dynamic_cast<const GeometryModel::Box<dim>*> (&this->get_geometry_model()))
+        {
+          const Point<dim> extents = gm->get_extents();
+
+          for (unsigned int d = 0; d < dim-1; ++d)
+            AssertThrow(representative_point[d] <= extents[d], ExcMessage("Coordinate "
+                                                                          + dealii::Utilities::int_to_string(d)
+                                                                          + " of representative point outside box domain."));
+
+          // set z of surface
+          representative_point[dim-1] = extents[dim-1];
+        }
+      else if (const GeometryModel::TwoMergedBoxes<dim> *gm = dynamic_cast<const GeometryModel::TwoMergedBoxes<dim>*> (&this->get_geometry_model()))
+        {
+          const Point<dim> extents = gm->get_extents();
+
+          for (unsigned int d = 0; d < dim-1; ++d)
+            AssertThrow(representative_point[d] <= extents[d], ExcMessage("Coordinate "
+                                                                          + dealii::Utilities::int_to_string(d)
+                                                                          +"Representative point outside box domain."));
+
+          // set z of surface
+          representative_point[dim-1] = extents[dim-1];
+        }
+      else
+        AssertThrow(false, ExcNotImplemented());
+
+      // set up the input for the density function of the material model
+      typename MaterialModel::Interface<dim>::MaterialModelInputs in0(1, n_compositional_fields);
+      typename MaterialModel::Interface<dim>::MaterialModelOutputs out0(1, n_compositional_fields);
+
+      // where to calculate the density
+      // for spherical domains
+      if (dynamic_cast<const GeometryModel::SphericalShell<dim>*> (&this->get_geometry_model()) != 0 ||
+          dynamic_cast<const GeometryModel::Sphere<dim>*> (&this->get_geometry_model()) != 0 ||
+          dynamic_cast<const GeometryModel::Chunk<dim>*> (&this->get_geometry_model()) != 0 ||
+          /*dynamic_cast<const GeometryModel::TwoMergedChunks<dim>*> (&this->get_geometry_model()) != 0*/
+          dynamic_cast<const GeometryModel::EllipsoidalChunk<dim>*> (&this->get_geometry_model()) != 0 )
+        {
+          // set spherical surface coordinates
+          in0.position[0] = spherical_to_cart(spherical_representative_point);
+        }
+      // and for cartesian domains
+      else
+        {
+          // set cartesian surface coordinates
+          in0.position[0] = representative_point;
+        }
+
+      // we need the actual temperature at t0
+      in0.temperature[0] = this->get_initial_conditions().initial_temperature(in0.position[0]);
+
+      // the previous pressure
```

Comment:
I guess this is not really the previous pressure, it is also the pressure at the surface for t=t0, so it might be easier just to write on comment that says we need the actual temperature, pressure and composition at the surface for t=t0.